### PR TITLE
fix(cloudflare-pages): Expose Cloudflare Pages type parameters

### PR DESCRIPTION
**Pull request by BarryThePenguin**
This PR aims to increase the type safety of the Cloudflare Pages handlers by exposing the type parameters of the `handle` and `handleMiddleware` functions. These types are really only useful when writing tests

### CODE REVIEW

File: src/adapter/cloudflare-pages/handler.test.ts [hunk 1]
```diff
@@ -2,30 +2,61 @@ import { getCookie } from '../../helper/cookie'
...
+      ASSETS: { fetch: vi.fn(), ...context.env?.ASSETS },
+      TOKEN: context.env?.TOKEN ?? 'HONOISCOOL',
+    },
+    functionPath: '_worker.js',
+    next: vi.fn(),
+    params: {},
+    passThroughOnException: vi.fn(),
+    request: new Request('http://localhost/api/foo'),
+    waitUntil: vi.fn(),
+    ...context,
```
Comment (BarryThePenguin) [inline src/adapter/cloudflare-pages/handler.test.ts:29]: Creating an `EventContext` in each test was getting quite repetitive, so I added this function that turns a `Partial<EventContext>` into an `EventContext` with all the necessary properties.

File: src/adapter/cloudflare-pages/handler.test.ts [hunk 2]
```diff
@@ -2,30 +2,61 @@ import { getCookie } from '../../helper/cookie'
...
+      request,
+      env,
+      waitUntil,
+      passThroughOnException,
+    })
     const app = new Hono<Env>()
+    const appFetchSpy = vi.spyOn(app, 'fetch')
     app.get('/api/foo', (c) => {
-      const reqInEventContext = c.env.eventContext.request
-      return c.json({ TOKEN: c.env.TOKEN, requestURL: reqInEventContext.url })
```
Comment (BarryThePenguin) [inline src/adapter/cloudflare-pages/handler.test.ts:23]: From my understanding, this was ensuring that the original `eventContext` was added to `c.env`

I replaced this by adding an assertion that `vi.spyOn(app, 'fetch')` was called with `{ ...env, eventContext }`

File: src/adapter/cloudflare-pages/handler.test.ts [hunk 3]
```diff
@@ -186,58 +192,81 @@ describe('Middleware adapter for Cloudflare Pages', () => {
...
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    await expect(handler({ request, env, next })).rejects.toThrowError('Something went wrong')
+    const eventContext = createEventContext({ request, next })
+    await expect(handler(eventContext)).rejects.toThrowError('Something went wrong')
     expect(next).not.toHaveBeenCalled()
   })
 })
+
+describe('serveStatic()', () => {
```
Comment (BarryThePenguin) [inline src/adapter/cloudflare-pages/handler.test.ts:235]: Also added test coverage for `serveStatic()`

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@BarryThePenguin 

The tests are cool! Thanks!

**Pull request merged**